### DISCUSSION TITLE
Parameterizing table names

### DISCUSSION THREAD

**Discussion by diamond29** [sqlc-dev/sqlc#1305] [Feature Requests & Ideas]
Is there a pattern or any interest in adding native support for parameterizing a table name?

In my database, we have a bunch of existing tables with fields like, start_at, end_at, and is_deleted. There are a lot of queries for different tables that look basically the same based on these fields, and they're repeated for sqlc in each file.

I considered moving this common data to another table, but that requires extra joins/queries at runtime, and for us to migrate our data. We could also write some code to generate the sqlc inputs, but that seems fragile.

Any other ideas?

**Comment by kyleconroy** [comment]
I'm wondering if we could take the same approach as `sqlc.arg`. What we really want here is a way to parameterize queries at generation time using templates. The issue with templates is that they eventually become complex and difficult to reason about ([just take a look at the templates we use to generate Go](https://github.com/kyleconroy/sqlc/blob/main/internal/codegen/golang/templates/stdlib/queryCode.tmpl)).

I'd like to either solve this via SQL or by using a real programming language for configuration (something like Skylark). I think a combination of YAML + templates won't scale for larger projecs.

> We could also write some code to generate the sqlc inputs, but that seems fragile.

Here's a placeholder Go template that does that.

```go
{{range .Tables}}
-- name: List{{.}} :many
SELECT start_at, end_at, and is_deleted FROM {{.}};
{{end}}
```

I agree that this feels fragile, but is likely the solution you'll need to go with in the short-term.
